lib/util/create-logs.js

```diff
@@ -3,74 +3,56 @@ const User = require('../models/User');
 const getPreferences = require('./user-preferences');
 const { getLocation, getPlaces } = require('./google-map-service');
 
-const randomIndex = arr => {
-    return Math.floor(Math.random() * (arr.length - 1));
-};
-
+const randomIndex = arr => Math.floor(Math.random() * (arr.length - 1));
 
-const getRandomType = types => {
-    return types[randomIndex(types)];
-};
-
-const getRandomPlace = place => {
-    return place[randomIndex(place)];
-};
+const getRandomItem = arr => arr[randomIndex(arr)];
 
+const toLog = userId => place => ({
+    place_id: place.place_id,
+    user: userId,
+    tags: place.tags,
+    name: place.name,
+    price: place.price || 2
+});
 
 const createNewLogs = (user, zipcode, keywords) => {
     return getLocation(zipcode)
-        .then(location => {
-            return Promise.all([Promise.resolve(location), getPreferences(user, keywords)]);
-        })
+        .then(location => Promise.all([Promise.resolve(location), getPreferences(user, keywords)]))
         .then(([location, userPrefs]) => {
-            return Promise.all(userPrefs.map(logPref => {
-                const type = getRandomType(logPref.types);
-                const keywords = logPref.keywords;
+            return Promise.all(userPrefs.map(({ types, keywords }) => {
+                const type = getRandomItem(types);
                 return getPlaces(location, type, keywords);
             }));
         })
         .then(logPlaces => {
-            const logsObj = logPlaces
+            return logPlaces
                 .filter(logPlace => logPlace.length > 0)
-                .map(getRandomPlace)
-                .map(logPlace => ({
-                    place_id: logPlace.placeId,
-                    user: user._id,
-                    tags: logPlace.types,
-                    name: logPlace.name,
-                    price: logPlace.price || 2
-                }));
-            return Log.create(logsObj);
+                .map(getRandomItem)
+                .map(toLog(user._id));
+        })
+        .then(Log.create);
+};
+
+const createPopularLogs = (user, zipcode, keywords) => {
```

Comment:
create a function for `createPopularLogs` to make the exported function more readable.

Improve some of the logic from the original implementation.